superset/connectors/sqla/models.py

```diff
@@ -262,47 +287,23 @@ def is_temporal(self) -> bool:
             return self.is_dttm
         return self.type_generic == GenericDataType.TEMPORAL
 
+    @property
+    def database(self) -> Database:
+        return self.table.database if self.table else self._database
+
     @property
     def db_engine_spec(self) -> Type[BaseEngineSpec]:
-        return self.table.db_engine_spec
+        return self.database.db_engine_spec
 
     @property
     def db_extra(self) -> Dict[str, Any]:
-        return self.table.database.get_extra()
+        return self.database.get_extra()
 
     @property
     def type_generic(self) -> Optional[utils.GenericDataType]:
         if self.is_dttm:
             return GenericDataType.TEMPORAL
 
-        bool_types = ("BOOL",)
```

Comment:
A nice byproduct of this change is this previous logic can be removed given that the `db_engine_spec` is accessible if associated with a `Query` or `SqlaTable`.